client/src/util/dataframe/summarize.js

```diff
@@ -1,40 +1,64 @@
 /*
 Private dataframe support functions
+
+TODO / XXX: for scalar/continuous data, this uses a naive method
+of computing quantiles.  Would be good to switch from sort to 
+partition at some point.
```

Comment:
partition is a partial sort.  At its simplest, to partition at the kth element means to arrange all element in an array such that the kth element is in the position it would be when the array is sorted, and all values less than that occur before the kth position, and all values sorting greater are after.

Example:  if you partition [3, 5, 0, 2, 10] on the 2nd position, it would [2, 0, 3, 10, 5], [0, 2, 3, 5, 10] or any other similar result.   Quicksort is essentially a series of partition operations, on smaller and smaller sub-arrays, until the entire array is sorted.

You can imaging doing this for 100 partitions, which would give you the percentiles.   For a big array, this would be substantially cheaper than sorting the entire array.

For reference, have a look at [numpy.partition](https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.partition.html)

Reply:
Ah! That makes sense now on how partition would replace sort.